lib/charms/loki_k8s/v0/loki_push_api.py

```diff
@@ -690,14 +690,17 @@ def __init__(
         super().__init__(charm, relation_name)
         self.charm = charm
         self._relation_name = relation_name
+        self.container = self.charm.unit.get_container("loki")
```

Comment:
In the charm.py `__init__` method you have:

```python
self._container = self.unit.get_container(self._name)
```

So you can access it from the `LokiPushApiProvider` object with:

```python
self.charm._container = self.unit.get_container(self._name)
```

Reply:
Both `self.charm._container` and `self.charm.unit.get_container("loki")` make inconvenient assumptions (charm must have a `_container` field; workload container name must be "loki").
I'll go with `self.charm._container` for now.